app/src/main/java/com/randomappsinc/padfriendfinder/Activities/TopLeadersActivity.java

```diff
@@ -0,0 +1,105 @@
+import com.randomappsinc.padfriendfinder.Adapters.TopMonsterAdapter;
+import com.randomappsinc.padfriendfinder.Misc.Constants;
+import com.randomappsinc.padfriendfinder.Misc.MonsterServer;
+import com.randomappsinc.padfriendfinder.Models.MonsterAttributes;
+import com.randomappsinc.padfriendfinder.Models.RestCallResponse;
+import com.randomappsinc.padfriendfinder.Models.TopLeader;
+import com.randomappsinc.padfriendfinder.R;
+
+import java.util.List;
+
+import butterknife.Bind;
+import butterknife.ButterKnife;
+import butterknife.OnItemClick;
+import de.greenrobot.event.EventBus;
+
+public class TopLeadersActivity extends AppCompatActivity {
+
+    Context context;
+    @Bind(R.id.topLeadersLV) ListView topLeadersLV;
+    @Bind(R.id.topLeadersPB) ProgressBar topLeadersPB;
+
+    TopMonsterAdapter topMonsterAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.top_leaders_layout);
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        ButterKnife.bind(this);
+        topMonsterAdapter = new TopMonsterAdapter(this);
+        context = this;
+        topLeadersPB.setVisibility(View.VISIBLE);
+        new GetTopLeaders(context).execute();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        EventBus.getDefault().register(this);
+    }
+
+    @Override
+    public void onStop() {
+        EventBus.getDefault().unregister(this);
+        super.onStop();
+    }
+
+    @OnItemClick(R.id.topLeadersLV)
+    public void onItemClick(AdapterView<?> adapterView, View view, final int position, long id) {
+        String name = topMonsterAdapter.getItem(position).getName();
+        MonsterAttributes monster = MonsterServer.getMonsterServer().getMonsterAttributes(name);
+        monster.setPlusEggs(Constants.MAX_PLUS_EGGS);
+        Intent intent = new Intent(context, FriendResultsActivity.class);
+        intent.putExtra(Constants.MONSTER_KEY, monster);
+        startActivity(intent);
+    }
+
+    public void onEvent(RestCallResponse restCallResponse) {
+        topLeadersPB.setVisibility(View.GONE);
+        if (restCallResponse.getStatusCode() == 200) {
+            Toast.makeText(context, Constants.FETCHED_TOP_LEADERS, Toast.LENGTH_SHORT).show();
+            List<TopLeader> topLeaderList = JSONParser.parseTopLeaders(restCallResponse.getResponse());
+            topMonsterAdapter.setTopMonsters(topLeaderList);
+            topLeadersLV.setAdapter(topMonsterAdapter);
+            topLeadersLV.setVisibility(View.VISIBLE);
+        }
+        else
+            Toast.makeText(context, Constants.FAILED_TO_FETCH, Toast.LENGTH_SHORT).show();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_top_leaders, menu);
```

Comment:
If this is just inflating a blank menu, inflate the blank_menu.xml I made and delete this new one.